packages/smart-wallet/src/smartWallet.js

```diff
@@ -954,7 +958,7 @@ export const prepareSmartWallet = (baggage, shared) => {
           const { zoe, agoricNames } = shared;
           const { invitationBrand, invitationIssuer } = shared;
 
-          facets.helper.assertUniqueOfferId(String(offerSpec.id));
+          facets.helper.assertUniqueOfferId(offerSpec.id);
```

Comment:
I was wondering if we should check whether the offerId parses as a number, and if it does, verify that no known offer use that ?

```suggestion
          facets.helper.assertUniqueOfferId(offerSpec.id);
          const maybeNumberId = Number.parseFloat(offerSpec.id);
          if (String(maybeNumberId) === offerSpec.id) {
            facets.helper.assertUniqueOfferId(maybeNumberId);
          }
```